src/frontends/tensorflow_common/src/op/size.cpp

```diff
@@ -14,13 +15,23 @@ namespace tensorflow {
 namespace op {
 
 ov::OutputVector translate_size_op(const NodeContext& node) {
-    auto data = node.get_input(0);
-    auto out_type = node.get_attribute<ov::element::Type>("out_type", ov::element::i32);
-    auto shape_of = make_shared<ShapeOf>(data, out_type);
-    auto axis = make_shared<Constant>(ov::element::i64, Shape{}, 0);
-    auto res = make_shared<ReduceProd>(shape_of, axis);
-    set_node_name(node.get_name(), res);
-    return res->outputs();
+    // Size operation computes a number of elements in the input tensor
+    default_op_checks(node, 1, {"Size"});
+    auto input = node.get_input(0);
+
+    // retrive attribute of the output type
+    auto out_type = node.get_attribute<element::Type>("out_type", element::i32);
+
+    // introduce extra dimension in order to compute size in case of a scalar input
+    auto const_zero = make_shared<Constant>(element::i32, Shape{1}, 0);
+    input = make_shared<Unsqueeze>(input, const_zero);
+
+    // compute the input tensor size
+    auto shape_of = make_shared<ShapeOf>(input, out_type);
+    auto axis = make_shared<Constant>(element::i32, Shape{}, 0);
```

Comment:
I don't get, should we use Shape{1} or we may use Shape[] instead. Could you take a look on this line before merge?